File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int map[] = new int[26];
        for(char ch:s1.toCharArray()) {
            map[ch-'a']++;
        }
        int left = 0;
        int right = 0;
        while(right < s2.length()) {
            if(map[s2.charAt(right)-'a'] > 0) {
                map[s2.charAt(right)-'a']--;
                if(right - left + 1 == s1.length()) return true;
                right++;
            }
            else {
                map[s2.charAt(left)-'a']++;
                left++;
            }
        }
        return false;
    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        count = [0 for i in range(26)]
        for ch in s1: count[ord(ch) - ord('a')] += 1
        left = 0
        right = 0
        while right < len(s2):
            if count[ord(s2[right]) - ord('a')] > 0:
                count[ord(s2[right]) - ord('a')] -= 1
                if right - left + 1 == len(s1): return True
                right += 1
            else:
                count[ord(s2[left]) - ord('a')] += 1
                left += 1
        return False